sources/packages/backend/apps/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -202,6 +197,11 @@ export class ApplicationStudentsController extends BaseController {
       );
     }
 
+    await this.applicationControllerService.offeringIntensityRestrictionCheck(
```

Comment:
Not part of the PR but a bit critical, if at line 174 the `getOfferingById` does not return an offering, we can throw as below:
```ts
if(!offering){
    throw new BadRequestException(
        "Selected offering id is invalid.",
    );
}
```